java/clients/src/main/java/sleeper/clients/QueryWebSocketClient.java

```diff
@@ -328,11 +365,13 @@ private void handleCompleted(JsonObject message, String queryId) {
         public void onClose(String reason) {
             out.println("Disconnected from WebSocket API: " + reason);
             queryComplete = true;
+            future.completeExceptionally(new WebSocketClosedException(reason));
```

Comment:
This looks incorrect? It looks like this will be called whether it was successful or not. It looks like this should only fail if the query didn't finish? Why isn't this failing a unit test?

Reply:
Thanks, I've added a check to see whether the code returned from onClose indicates it was closed normally or not, and added more close calls in tests